### PR TITLE
Add DKG support to the manifest

### PR DESCRIPTION
**Pull request by ejfitzgerald**

### CODE REVIEW

File: libs/shards/tests/unit/manifest_parse_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    "dkg": { "uri": "tcp://127.0.0.1:8002", "port": 50002},
+    "lanes": [
+        { "uri": "tcp://127.0.0.1:8010", "port": 50010},
+        { "uri": "tcp://127.0.0.1:8012", "port": 50012}
+    ]
+}
+)";
+
+  Manifest manifest{};
+  ASSERT_TRUE(manifest.Parse(FULL_MANIFEST));
```
Comment (wkaluza) [inline libs/shards/tests/unit/manifest_parse_tests.cpp:46]: Please move to fixture; split this into five tests, as indicated by comments.

File: libs/shards/tests/unit/manifest_entry_tests.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+  Address const address{"<muddle address>"};
+  entry.UpdateAddress(address);
+  EXPECT_EQ(entry.address(), address);
+}
+
+TEST(ManifestEntryTests, CheckUriWithLocalPortConstruction)
+{
+  Uri           uri{"tcp://127.0.0.1:1234"};
+  ManifestEntry entry{uri, 4321};
```
Comment (wkaluza) [inline libs/shards/tests/unit/manifest_entry_tests.cpp:72]: Curious: what use case does this represent? As in, when would we want to provide a URI with one port and then override it? in the manifest with another?

**Review by wkaluza — CHANGES_REQUESTED** [review]
Looks ok except for test layout. Should be good to go once static analysis is passing.

File: libs/shards/tests/unit/manifest_entry_tests.cpp (hunk 2 above)
Comment (ejfitzgerald) [reply libs/shards/tests/unit/manifest_entry_tests.cpp:72]: It is covering the use case where you bind to a local port, but announce the connection on another point

File: libs/shards/tests/unit/manifest_parse_tests.cpp (hunk 1 above)
Comment (ejfitzgerald) [reply libs/shards/tests/unit/manifest_parse_tests.cpp:46]: Lets discuss

**Review by wkaluza — APPROVED** [review]

**Pull request merged**